migrations/6_update_dates_existing_published_projects.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+exports.up = function (knex, Promise) {
+  return knex('publishedProjects').select('id', 'date_created', 'date_updated')
+  .then(function(publishedProjects) {
+    return Promise.map(publishedProjects, function(publishedProject) {
+      var publishedProjectId = publishedProject.id;
+      var dateCreated = publishedProject.date_created;
+      var dateUpdated = publishedProject.date_updated;
+
+      // For projects published after date tracking was added
+      if (dateCreated && dateUpdated) {
+        return;
+      }
+
+      // For existing published projects that have been updated since
+      // date tracking has been added, they will have a date_updated
+      // but not a date_created. Set the date_created to the date_updated
+      if (dateUpdated) {
+        return knex('publishedProjects').update('date_created', dateUpdated)
+        .where('id', publishedProjectId);
+      }
+
+      // For existing published projects that have not been updated
+      // since date tracking has been added - set both the dates
+      // to the date_updated of the corresponding project
+      return knex('projects')
+      .where('published_id', publishedProjectId).select('date_updated')
+      .then(function(projects) {
+        var date = projects[0].date_updated;
+
+        return Promise.join(
```

Comment:
Can you combine these two queries together somehow? I don't know the knex API well enough, but in SQL it'd be something like `UPDATE publishedProjects SET date_created = $1, date_updated = $2 WHERE id = $3`